platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/LongListHeap.java

```diff
@@ -87,23 +93,19 @@ public LongListHeap(final Path file, final Configuration configuration) throws I
 
     /** {@inheritDoc} */
     @Override
-    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
-        // read data
-        final int numOfArrays = calculateNumberOfChunks(size());
-        final ByteBuffer buffer = allocateDirect(memoryChunkSize);
-        buffer.order(ByteOrder.nativeOrder());
-        for (int i = 0; i < numOfArrays; i++) {
-            final AtomicLongArray atomicLongArray = new AtomicLongArray(numLongsPerChunk);
-            buffer.clear();
-            MerkleDbFileUtils.completelyRead(fileChannel, buffer);
-            buffer.flip();
-            int index = 0;
-            while (buffer.remaining() > 0) {
-                atomicLongArray.set(index, buffer.getLong());
-                index++;
-            }
-            chunkList.set(i, atomicLongArray);
+    protected void readChunkData(FileChannel fileChannel, int chunkIndex, int startIndex) throws IOException {
+        AtomicLongArray chunk = createChunk();
+
+        ByteBuffer buffer = TEMP_READ_BUFFER_THREAD_LOCAL.get();
+        MerkleDbFileUtils.completelyRead(fileChannel, buffer);
```

Comment:
It would read the whole buffer, which is DEFAULT_NUM_LONGS_PER_CHUNK * 8 bytes. Instead, it should only read memoryChunkSize - startIndex * 8 bytes.